pkg/common/utils/utils.go

```diff
@@ -44,6 +45,10 @@ const userInfoKey = siCommon.DomainYuniKorn + "user.info"
 const uniqueAutogenSuffix = "-uniqueautogen"
 
 var pluginMode bool
+var (
+	// ErrorTimeout returned if waiting for a condition times out
+	ErrorTimeout = errors.New("timeout waiting for condition")
```

Comment:
this can't be constant, so maybe we can focus on "error message" instead of `Error`. For example:
```go
const timeoutErrorMessage="timeout waiting for condition"
```

and then in testing we can check the error message instead of "error"

Reply:
OK. I will use `ErrorContains `to check it.
In core repo, `WaitForCondition `unit test have implemented with the same usage.
I will create a issue to improve it.

Reply:
Could have a look for this?
I think we should have a consistent way.

https://github.com/apache/yunikorn-core/pull/875#discussion_r1621576743

Reply:
ok, let's keep the consistency